script.js

```diff
@@ -35,9 +35,44 @@ function makeEpisodeCard(episode) {
   episodeCard.querySelector(".episode-card__img").src = episodeImage;
   episodeCard.querySelector(".episode-card__summary").outerHTML = episodeSummary;
 
-
   return episodeCard;
 }
 
+// This parts is for the search bar and filtering through episodes
+function setupSearch() {
+  const searchInput = document.getElementById("search-input");
+  searchInput.addEventListener("input", function () {
+    const searchTerm = searchInput.value.trim().toLowerCase();
+    filterEpisodes(searchTerm);
+  });
+}
```

Comment:
I like how you've separated the search logic into its own function. This helps keep the code modular and easier to maintain. Breaking it down further into smaller functions is also a good practice.